src/client/components/ImageSetter/ImageSetter.js

```diff
@@ -25,11 +62,58 @@ const ImageSetter = ({
         name: filename,
         id: uuidv4(),
       };
-      handleAddImageByLink(newImage);
+      const img = new Image();
+      img.src = newImage.src;
+      img.onload = () => checkIsImage(true, newImage);
+      img.onerror = () => checkIsImage(false, newImage);
       imageLinkInput.current.value = '';
     }
   };
 
+  const disableAndInsertImage = (image, imageName = 'image') => {
+    const newImage = {
+      src: image,
+      name: imageName,
+      id: uuidv4(),
+    };
+    if (isMultiple) {
+      setCurrentImages([...currentImages, newImage]);
+    } else setCurrentImages([newImage]);
+    setLoadingImage(false);
+    onLoadingImage(false);
+  };
+
+  const handleChangeImage = e => {
+    if (e.target.files && e.target.files[0]) {
+      if (
+        !isValidImage(e.target.files[0], MAX_IMG_SIZE[objectFields.background], ALLOWED_IMG_FORMATS)
+      ) {
+        onImageInvalid(
+          MAX_IMG_SIZE[objectFields.background],
+          `(${ALLOWED_IMG_FORMATS.join(', ')}) `,
+        );
+        return;
+      }
+
+      if (!isMultiple) {
+        setCurrentImages([]);
+      }
+      setLoadingImage(true);
+      onLoadingImage(true);
+
+      onImageUpload(e.target.files[0], disableAndInsertImage, () => {
+        setLoadingImage(false);
+        onLoadingImage(false);
+      });
+    }
+  };
+
+  const handleRemoveImage = imageId => {
+    const filteredImages = currentImages.filter(f => f.id !== imageId);
```

Comment:
why 'f' when we filter images? 'img => img.id !== imageId' looks better

Reply:
++